lib/NamedModulesPlugin.js

```diff
@@ -12,13 +21,36 @@ class NamedModulesPlugin {
 	apply(compiler) {
 		compiler.hooks.compilation.tap("NamedModulesPlugin", compilation => {
 			compilation.hooks.beforeModuleIds.tap("NamedModulesPlugin", modules => {
+				const namedModules = new Map();
+
 				for (const module of modules) {
 					if (module.id === null && module.libIdent) {
 						module.id = module.libIdent({
 							context: this.options.context || compiler.options.context
 						});
 					}
+
+					if (module.id !== null) {
+						if (namedModules.has(module.id)) {
+							namedModules.get(module.id).push(module);
+						} else {
+							namedModules.set(module.id, [module]);
+						}
+					}
 				}
+
+				namedModules.forEach(namedModule => {
+					if (namedModule.length > 1) {
+						namedModule.forEach(module => {
+							if (module.issuer && module.issuer.id) {
+								const requestShortener = new RequestShortener(module.context);
```

Comment:
Passing `this.options.context || compiler.options.context` instead.

As this value is already used in the code above, extract it into a const variable.